src/MongoDB.Driver/MongoCollectionImpl.cs

```diff
@@ -977,8 +976,8 @@ private IBsonSerializer<TField> GetValueSerializerForDistinct<TField>(RenderedFi
 
         private TResult ExecuteReadOperation<TResult>(IClientSessionHandle session, IReadOperation<TResult> operation, CancellationToken cancellationToken = default(CancellationToken))
         {
-            var readPreference = GetReadPreference(session);
-            return ExecuteReadOperation(session, operation, readPreference, cancellationToken);
+            var effectiveReadPreference = ReadPreferenceResolver.GetEffectiveReadPreference(session, null, _settings.ReadPreference);
```

Comment:
Note that the explicitReadPreference parameter is null, and the defaultReadPreference parameter is the ReadPreference from the MongoCollectionSettings.